### ISSUE TITLE
unused_closure_parameter should apply to closure declarations

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1082]
Currently, `unused_closure_parameter` rule is only applied for closures used in method calls. It should be applied to closure declarations as well:

```swift
let failure: Failure = { task, error in
    observer.sendFailed(error)
}
```
